edb/pgsql/compiler/dml.py

```diff
@@ -749,9 +707,75 @@ def dynamic_get_path(
     ctx.toplevel_stmt.append_cte(contents_cte)
     contents_rvar = relctx.rvar_for_rel(contents_cte, ctx=ctx)
 
+    rewrites = ir_stmt.rewrites and ir_stmt.rewrites.by_type.get(typeref)
+
+    pol_expr = ir_stmt.write_policies.get(typeref.id)
+    pol_ctx = None
+    if pol_expr or rewrites:
+        with ctx.new() as pol_ctx:
```

Comment:
What's going on here?

Reply:
I'll add a comment; I experimented with adding a function but that seems more confusing